testsuite/shared/src/main/java/org/wildfly/test/jmx/ControlledStateNotificationListener.java

```diff
@@ -82,6 +83,20 @@ private void writeNotification(AttributeChangeNotification notification, Path pa
             in.write(String.format("%s %s %s %s", notification.getType(), notification.getSequenceNumber(), notification.getSource().toString(), notification.getMessage()));
             in.newLine();
             in.flush();
+        } catch (ClosedByInterruptException cbie) {
+            /* This happens sometimes during the "reload" operation, the BufferedWriter won't write anymore
+               because our thread was interrupted by the server.
+               So we clear the interruption flag and try writing again and hope that it will work this time.
+            */
+            Thread.interrupted();
```

Comment:
If this returns true you should interrupt the thread again in a finally block. The server wanted the thread interrupted so that should be respected.